File: src/C#/Parcial2/ActividadP2A4/TrabajadorPiezas.cs
using System;

public class TrabajadorPiezas : Empleado
{
    private double salario;
    private int piezas;

    public TrabajadorPiezas(string nombre, string apellido, string nss, double salarioBase, int piezasHechas)
        : base(nombre, apellido, nss)
    {
        EstablecerSalario(salarioBase);
        EstablecerPiezas(piezasHechas);
    }

    public void EstablecerSalario(double salarioBase) { salario = (salarioBase < 0.0) ? 0.0 : salarioBase; }
    public double ObtenerSalario() { return salario; }
    public void EstablecerPiezas(int piezasHechas) { piezas = (piezasHechas < 0) ? 0 : piezasHechas; }
    public int ObtenerPiezas() { return piezas; }

    public override double Ingresos() { return ObtenerSalario() * ObtenerPiezas(); }

    public override string ToString()
    {
        return "trabajador por piezas: " + base.ToString() + "\nsalario por pieza: $" + ObtenerSalario() + "; piezas producidas: " + ObtenerPiezas();
    }
}

File: src/Java/Parcial2/ActividadP2A4/TrabajadorPiezas.java
public class TrabajadorPiezas extends Empleado {
   private double salario;
   private int piezas;

   public TrabajadorPiezas(String nombre, String apellido, String nss, double salarioBase, int piezasHechas) {
      super(nombre, apellido, nss);
      establecerSalario(salarioBase);
      establecerPiezas(piezasHechas);
   }

   public void establecerSalario(double salarioBase) {
      salario = (salarioBase < 0.0) ? 0.0 : salarioBase;
   }

   public double obtenerSalario() {
      return salario;
   }

   public void establecerPiezas(int piezasHechas) {
      piezas = (piezasHechas < 0) ? 0 : piezasHechas;
   }

   public int obtenerPiezas() {
      return piezas;
   }

   public double ingresos() {
      return obtenerSalario() * obtenerPiezas();
   }

   public String toString() {
      return String.format("%s: %s\n%s: $%,.2f; %s: %d", 
         "trabajador por piezas", super.toString(), 
         "salario por pieza", obtenerSalario(), 
         "piezas producidas", obtenerPiezas());
   }
}
